lib/cn/cn.c

```diff
@@ -414,10 +414,17 @@ cn_mblocks_commit(
     *n_committed = 0;
 
     for (i = 0; i < num_lists; i++) {
-        /* If we have a valid block ID. This check is protected in
-         * commit_mblock() when using a for loop where blk_list::n_blks is the
-         * upper bound. If there are no blocks in the block list, no blocks are
-         * committed.
+        /* The order of committing here, hblock, kblocks, and then vblocks is
+         * sensitive due to the n_committed argument. This argument is passed to
+         * cn_mblocks_destroy() to determine which blocks need to be aborted or
+         * deleted.
+         */
+
+        /* This check is similar to the check in cn_commit_blks() where the
+         * bounds on the for loop uses the block count of the block list. Here
+         * we always have at least 1 hblock to validate, and we do that by
+         * checking to make sure the hblock's block ID is valid prior to
+         * committing it.
          */
```

Comment:
Can we ask mpool the status of an mblock and then call abort or delete as needed?  Then we wouldn't need this goofball n_committed logic.

Reply:
Not currently possible. Mpool is not tracking the state of an mblock. @nabeelmmd and I discussed this earlier. Nabeel, can comment more.

Reply:
That makes me wonder what happens if mblocks are misused.  For example:
- write to a committed mblock
- abort a committed mblock
- delete an allocated but not committed mblock

do these operations fail?

Reply:
Yes, the only thing that's tracked in-memory for an mblock is its write length. If there's a pressing need from cN then we can add logic in mpool to determine whether an mblock is committed or not by reading its metadata file.